### PR TITLE
Syncing of rich user profile

### PR DESCRIPTION
**Pull request by vytis**
### Changes

Added syncing of user rich profile with `UserRichProfileRequestStrategy`.

## Dependencies

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-data-model/pull/687
- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/117
- [x] https://github.com/wireapp/wire-ios-mocktransport/pull/110

### CODE REVIEW

File: Tests/Source/Data Model/ServiceUserTests.swift [hunk 1]
```diff
@@ -22,6 +22,39 @@ import XCTest
...
+    var expiresAfter: TimeInterval = 0
+    
+    var readReceiptsEnabled: Bool = true
+    
+    var richProfile: [UserRichProfileField] = []
+    
+    var canCreateConversation: Bool = true
+    
+    func canAccessCompanyInformation(of user: UserType) -> Bool {
+        return true
```
Comment (alexaubry) [inline Tests/Source/Data Model/ServiceUserTests.swift]: Bots cannot see the user's company information.

```suggestion
        return false
```

File: Tests/Source/Data Model/ServiceUserTests.swift [hunk 2]
```diff
@@ -22,6 +22,39 @@ import XCTest
...
+    }
+    
+    func canAddUser(to conversation: ZMConversation) -> Bool {
+        return true
```
Comment (alexaubry) [inline Tests/Source/Data Model/ServiceUserTests.swift]: Bots cannot add users.

```suggestion
        return false
```

File: Tests/Source/Data Model/ServiceUserTests.swift [hunk 3]
```diff
@@ -22,6 +22,39 @@ import XCTest
...
+    }
+    
+    func canRemoveUser(from conversation: ZMConversation) -> Bool {
+        return true
```
Comment (alexaubry) [inline Tests/Source/Data Model/ServiceUserTests.swift]: Bots cannot remove users.

```suggestion
        return false
```

File: Tests/Source/Data Model/ServiceUserTests.swift [hunk 4]
```diff
@@ -22,6 +22,39 @@ import XCTest
...
+    
+    var isWirelessUser: Bool = false
+    
+    var expiresAfter: TimeInterval = 0
+    
+    var readReceiptsEnabled: Bool = true
+    
+    var richProfile: [UserRichProfileField] = []
+    
+    var canCreateConversation: Bool = true
```
Comment (alexaubry) [inline Tests/Source/Data Model/ServiceUserTests.swift]: Bots cannot create conversations.

```suggestion
    var canCreateConversation: Bool = false
```

**Comment by vytis** [comment]
@zenkins release major

**Pull request merged**